### PR TITLE
update Windows-specific API list

### PR DESCRIPTION
**Pull request by adamsitnik**
while working on https://github.com/dotnet/runtime/pull/40610 I've found some minor issues

@terrajobst @jeffhandley PTAL

### CODE REVIEW

File: accepted/2020/windows-specific-apis/windows-specific-apis.md [hunk 1]
```diff
@@ -133,8 +133,6 @@ Windows-specific:
 ### System.Diagnostics
 
 * **Process**
-    - `EnterDebugMode()`
-    - `LeaveDebugMode()`
```
Comment (adamsitnik) [inline accepted/2020/windows-specific-apis/windows-specific-apis.md:137]: these two methods are `nop` on non-Windows, some time ago @stephentoub has suggested to not to mark `nop` are Windows-specific

File: accepted/2020/windows-specific-apis/windows-specific-apis.md [hunk 2]
```diff
@@ -177,8 +175,6 @@ Windows-specific:
 
 * **NamedPipeClientStream**
     - `get_NumberOfServerInstances()`
-* **NamedPipeServerStream**
-    - `WaitForConnection()`
```
Comment (adamsitnik) [inline accepted/2020/windows-specific-apis/windows-specific-apis.md:181]: it's supported on Unix: https://github.com/dotnet/runtime/blob/08643df14054d1273002254c909fa099649ebfcf/src/libraries/System.IO.Pipes/src/System/IO/Pipes/NamedPipeServerStream.Unix.cs#L54-L66

File: accepted/2020/windows-specific-apis/windows-specific-apis.md [hunk 3]
```diff
@@ -245,11 +241,7 @@ Windows-specific:
     - `AddEventHandler(Object, Delegate)`
     - `RemoveEventHandler(Object, Delegate)`
 * **ComEventsHelper**
-    - `Combine(Object, Guid, Int32, Delegate)`
-    - `Remove(Object, Guid, Int32, Delegate)`
```
Comment (adamsitnik) [inline accepted/2020/windows-specific-apis/windows-specific-apis.md:249]: all methods of this type are Windows-specific so we should most probably make the entire type as such

https://github.com/dotnet/runtime/blob/e7204f5d6fcaca5e097ec854b3be6055229fc442/src/libraries/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs#L224-L228

File: accepted/2020/windows-specific-apis/windows-specific-apis.md [hunk 4]
```diff
@@ -245,11 +241,7 @@ Windows-specific:
...
 * **DispatchWrapper**
-    - `.ctor(Object)`
-    - `get_WrappedObject()`
```
Comment (adamsitnik) [inline accepted/2020/windows-specific-apis/windows-specific-apis.md:252]: all methods of this type are Windows-specific so we should most probably make the entire type as such

https://github.com/dotnet/runtime/blob/e7204f5d6fcaca5e097ec854b3be6055229fc442/src/libraries/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs#L317-L322

File: accepted/2020/windows-specific-apis/windows-specific-apis.md [hunk 5]
```diff
@@ -314,7 +306,6 @@ Windows-specific:
 
 * **X509Chain**
     - `.ctor(IntPtr)`
-* **X509Certificate2UI**
```
Comment (adamsitnik) [inline accepted/2020/windows-specific-apis/windows-specific-apis.md:317]: it's definied in `System.Windows.Extensions.dll` which is marked as Windows-specific

https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Windows.Extensions/ref/System.Windows.Extensions.cs#L121

File: accepted/2020/windows-specific-apis/windows-specific-apis.md [hunk 6]
```diff
@@ -439,16 +430,6 @@ Windows-specific:
...
-### System.Xaml.Permissions
-
-* **XamlAccessLevel**
-    - `AssemblyAccessTo(Assembly)`
-    - `AssemblyAccessTo(AssemblyName)`
-    - `get_AssemblyAccessToAssemblyName()`
-    - `get_PrivateAccessToTypeName()`
-    - `PrivateAccessTo(String)`
-    - `PrivateAccessTo(Type)`
-
```
Comment (adamsitnik) [inline accepted/2020/windows-specific-apis/windows-specific-apis.md:451]: it's defined in `System.Windows.Extensions.dll` which is marked as Windows-specific

https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Windows.Extensions/ref/System.Windows.Extensions.cs#L135-L147

File: accepted/2020/windows-specific-apis/windows-specific-apis.md (hunk 5 above)
Comment (terrajobst) [reply accepted/2020/windows-specific-apis/windows-specific-apis.md:317]: Good catch!

File: accepted/2020/windows-specific-apis/windows-specific-apis.md (hunk 2 above)
Comment (terrajobst) [reply accepted/2020/windows-specific-apis/windows-specific-apis.md:181]: Makes sense

File: accepted/2020/windows-specific-apis/windows-specific-apis.md (hunk 1 above)
Comment (terrajobst) [reply accepted/2020/windows-specific-apis/windows-specific-apis.md:137]: Makes sense.

**Pull request merged**